_posts/2020-02-06-pwa-cache-me-if-you-can.markdown

```diff
@@ -0,0 +1,245 @@
+---
+layout: post
+title: "PWA Cache me if you can"
+date: 2020-02-06
+tags: PWA react
+author: Natalie Laing
+image: img/code.jpg
+---
+
+
+### What is a PWA anyway?
+
+> This morning I went to write an article outlining a few tips about implementing a progressive web app (PWA). But when I went to introduce the topic, I again encountered what was the hardest part about the whole topic in the first place: I have a really hard time describing what a progressive web app actually is.
+
+
+![in the feels](/img/blog/pwa/feels.jpeg)
+
+The term PWA was coined in 2015, by designer Frances Berriman and Google Chrome engineer Alex Russell. 
+
+To create a PWA, you are going to need a service worker. A service worker is a script that is run in the background and acts as a network proxy between your app and the server to deliver that efficient caching strategy and offline capability PWA's are known for. 
+Your request will go to your service worker; the service worker will check if what you are requesting is available in the cache. If it can extract the request data from the cache otherwise it will send the request to the server.
+
+A PWA will require a manifest file that details how the app should behave when installed such as what name and icon will be displayed to the user.
+
+![service worker](/img/blog/pwa/service-worker.png)
+
+### But why do I need one? 
+
+Handy checklist:
+
+* **Do you require offline access?** 
+* You can cache important assets and serve them to the user to view when they are offline or have inadequate internet connection.
+* So your app requires users to upload files or images but they lose connection, your PWA can save those files locally and send them to the server once you reconnect to the internet.
+
+* **Is security important?**
+* a requirement of PWA's is they must be served over HTTPS.
+
+* **Do you want faster load times?**
+* Do you care about your users? Then load times will be extremely important to keep your users on your PWA.
+* You want that high-performance audit score.
+
+* **Do you want your users to be able to install your PWA to their home screen?**
+
```

Comment:
Might be worth including something about it being their _mobile_ home screen or desktop